src/views/notification/components/NotificationList/index.tsx

```diff
@@ -0,0 +1,69 @@
+import { Box, Divider } from '@mui/material'
+import { styled } from '@mui/material/styles'
+import { Children, FC } from 'react'
+import LoadingScreen from '~/components/LoadingScreen'
+import { NotificationRecipientResponse } from '~/features/notification/types/notificationRecipient'
+import { Pagination } from '../../types'
+import EmptyState from '../EmptyState'
+import NotificationCard from '../NotificationCard'
+import Header from './components/Header'
+
+// -------------------------- types --------------------------
+
+type NotificationListProps = {
+  isLoading: boolean
+  pagination: Pagination
+  isReadChecking: boolean
+  data: NotificationRecipientResponse | undefined
+}
+
+// -------------------- styled components --------------------
+
+const WrapperList = styled(Box)(({ theme }) => ({
+  overflow: 'hidden',
+  border: '1px solid',
+  borderRadius: '8px',
+  borderColor: theme.palette.neutral700.main,
+  backgroundColor: theme.palette.neutral900.main
+}))
+
+// ----------------------- components ------------------------
+
+const NotificationList: FC<NotificationListProps> = ({
+  data,
+  pagination,
+  isLoading,
+  isReadChecking
+}) => {
+  const { count, rows: notifications } = data || {}
+
+  return (
+    <WrapperList>
+      <Header
+        count={count || 0}
+        pagination={pagination}
+        isReadChecking={isReadChecking}
+        dataLength={notifications?.length || 0}
+      />
+
+      <Divider sx={{ borderColor: 'neutral700.main' }} />
+
+      {isLoading ? (
+        <LoadingScreen />
+      ) : count ? (
+        Children.toArray(
+          notifications?.map((notification, index) => (
+            <NotificationCard
+              data={notification}
+              lastIndex={index === notifications.length - 1}
+            />
+          ))
+        )
+      ) : (
+        <EmptyState />
+      )}
```

Comment:
```suggestion
     const Notifications = () => {
         if(count){
            return Children.toArray(
          notifications?.map((notification, index) => (
            <NotificationCard
              data={notification}
              lastIndex={index === notifications.length - 1}
            />
          ))
         }
         return <EmptyState />
     }


      {isLoading ? (
        <LoadingScreen />
      ) : 
        <Notifications />
      }
```

hindari nested tenary ya. biar readability dan maintainability nya lebih mantep